src/link.js

```diff
@@ -0,0 +1,24 @@
+
+import { Inertia, shouldIntercept } from '@inertiajs/inertia'
+
+
+export default function inertia(node, options) {
+
+  function handleClick(event) {
+    if (shouldIntercept(event)) {
+      event.preventDefault()
+      Inertia.visit(node.href, options)
+    }
+  }
```

Comment:
Yes, I understand that. But then you are not automatically using the default settings for preserveState. What I was thinking is that one could pass the type of Inertia Request to the action and it would use the defaults.
Something like this.

```suggestion
export default function inertia(node, options = {}) {

  function handleClick(event) {
    if (shouldIntercept(event)) {
      event.preventDefault()

      const { type = 'visit', data = {}, options: settings = {} } = options

      if (
        type === 'visit'
        || type === 'replace'
        || type === 'delete'
      ) {
        Inertia[type](node.href, settings)
      } else if (type === 'reload') {
        Inertia[type](settings)
      } else {
        Inertia[type](node.href, data, settings)
      }
    }
  }
```
Then you could use it like this:
```svelte
<a href="/" use:inertia>Link Text</a>
<a href="/items/delete/1" use:inertia={{ type: 'delete' }}>Link Text</a>
<a href="/" use:inertia={{ type: 'post', data: { foo: 'bar' } }}>Link Text</a>
<a href="/items" use:inertia={{ type: 'reload', options: { only: ['items'] } }}>Link Text</a>
```